velox/exec/Window.cpp

```diff
@@ -471,14 +496,46 @@ void Window::callApplyLoop(
   }
 }
 
+RowVectorPtr Window::createFinalOutput() {
+  if (output_->size() > numRowsPerOutput_) {
+    auto result = std::dynamic_pointer_cast<RowVector>(
+        output_->slice(0, numRowsPerOutput_));
+    auto remainingSize = output_->size() - numRowsPerOutput_;
+    output_ = std::dynamic_pointer_cast<RowVector>(
+        output_->slice(numRowsPerOutput_, remainingSize));
+    return result;
+  } else if (noMoreInput_) {
+    auto result = std::move(output_);
+    return result;
+  } else {
+    return nullptr;
+  }
+}
+
+RowVectorPtr Window::getWindowOutput(vector_size_t numOutputRows) {
+  auto result = std::dynamic_pointer_cast<RowVector>(
+      BaseVector::create(outputType_, numOutputRows, operatorCtx_->pool()));
+
+  for (int i = numInputColumns_; i < outputType_->size(); i++) {
+    auto output = BaseVector::create(
+        outputType_->childAt(i), numOutputRows, operatorCtx_->pool());
+    result->childAt(i) = output;
+  }
+
+  // Compute the output values of window functions.
+  callApplyLoop(numOutputRows, result);
+
+  return result;
+}
+
 RowVectorPtr Window::getOutput() {
```

Comment:
@aditi-pandit  I have refactored the StreamingWindow based on https://github.com/facebookincubator/velox/pull/6011. And the `getOutput ` operator differs between `StreamingWindow `and `SortWindow`. Perhaps we can create a `WindowOutput ` class to distinctively handle `StreamingOutput `and `SortWindowOutput` , similar to WindowBuild. What are your thoughts on this approach?

Reply:
This getOutput change should not be needed. https://github.com/facebookincubator/velox/pull/6212 should work I think.

Reply:
It seems https://github.com/facebookincubator/velox/pull/6212 not work in StreamingWindowBuild use case.  It will cause the wrong result number.